src/server.cpp

```diff
@@ -127,6 +127,98 @@ void *ServerThread::run()
 	return nullptr;
 }
 
+class MapThread : public Thread
+{
+public:
+
+	MapThread(Server *server):
+		Thread("Server"),
+		m_server(server)
+	{}
+
+	void *run();
+
+private:
+	Server *m_server;
+};
+
+void *MapThread::run()
+{
+	BEGIN_DEBUG_EXCEPTION_HANDLER
+
+	static thread_local const float save_interval =
+		g_settings->getFloat("server_map_save_interval") * 1000;
+
+	u64 last_time = 0;
+	int flush_counter = 0;
+	while (!stopRequested()) {
+		sleep_ms(100); // no busy loop
+
+		u64 curr_time = porting::getTimeMs();
+		if (curr_time - last_time < save_interval)
+			continue;
+
+		last_time = curr_time;
+
+		ServerEnvironment &env = m_server->getEnv();
+		ServerMap &map = (ServerMap&) env.getMap();
+
+		ModifiedState state;
+		if (++flush_counter < 4) {
+			infostream << "Saving Map" << std::endl;
+			state = MOD_STATE_WRITE_NEEDED;
+		} else {
+			infostream << "Flushing Map" << std::endl;
```

Comment:
Happy to remove this part. The intent is to periodically even save these blocks to that server shutdown is faster.
Might be misguided.